### ISSUE TITLE
Generalize the use of links to the theory documentation from the API one

### ISSUE THREAD

**Issue by mdelozzo** [openturns/openturns#2114]
Hello, 

And thank you very much for your great library! My first contribution will not be a proposal of functionality but a remark on the documentation.

In the theory documentation, the pages propose links to the corresponding API documentation but unfortunately the opposite is not always true. Moreover, it is not easy to guess the path to the page of a documentation corresponding to the page of the other documentation as their tables of contents are not mirrors of each other ([theory TOC](http://openturns.github.io/openturns/latest/theory/theory.html) / [API TOC](http://openturns.github.io/openturns/latest/user_manual/user_manual.html))

Being able to move easily from theory to API *and* from API to theory is really a good thing and I think it should be generalized.

Additional remark: the theory related to some mathematica objects is available in API pages but is missing in the theory documentation. It can sometimes be confusing to discover concepts, algorithms, ... in the API. Why not put all the theory materials in the theory documentation and use reminder in some API pages if necessary?

Some examples:

- the [theory documentation of copulas](http://openturns.github.io/openturns/latest/theory/probabilistic_modeling/copulas.html) refers to [the API of the available copulas](http://openturns.github.io/openturns/latest/user_manual/probabilistic_modelling.html#copulas) but [the API of the normal copula](http://openturns.github.io/openturns/latest/user_manual/_generated/openturns.NormalCopula.html#openturns.NormalCopula) does not refer to the theory of copulas. Furthermore, the theory of normal copula is documented in a **Notes** section of the API page but no information can be found about the normal copula in the theory documentation.
- the [theory documentation of Kriging](http://openturns.github.io/openturns/latest/theory/meta_modeling/kriging.html) refers to [the API documentation of Kriging](http://openturns.github.io/openturns/latest/user_manual/response_surface/_generated/openturns.KrigingAlgorithm.html#openturns.KrigingAlgorithm) and vice versa.
- [the theory documentation of HSIC](http://openturns.github.io/openturns/latest/theory/reliability_sensitivity/sensitivity_hsic.html) refers to [the API documentation](http://openturns.github.io/openturns/latest/user_manual/_generated/openturns.HSICEstimator.html?highlight=hsic#openturns.HSICEstimator) but the opposite is false.

**Comment by mbaudin47** [comment]
Thank you very much for this issue!
We analyze your examples and have a suggestion to improve the current state. We noticed that the way that current API help pages present links to the theory help page is somewhat inconsistent. For example, see the "Refer to" in the following help page [FittingTest.AIC](https://openturns.github.io/openturns/latest/user_manual/_generated/openturns.FittingTest.AIC.html):

![image](https://user-images.githubusercontent.com/31351465/190431684-c57667e8-1d65-4189-8340-1fccc0a845b9.png)

In the [theory document of Kriging](https://openturns.github.io/openturns/latest/theory/meta_modeling/kriging.html), we see links from the theory to the API and examples:

![image](https://user-images.githubusercontent.com/31351465/190430471-7fd3fe00-39d8-4ed5-9799-ce3d860c173f.png)

Would you agree that a good practice would be to include a new category within API help pages, e.g.:

```
Theory:
     - Link 1
     - Link 2
```

**Comment by mdelozzo** [comment]
Yes, I would. Your proposal seems to be a good improvement. Thanks!
